src/shared/components/ui/Dropdown/Dropdown.js

```diff
@@ -62,11 +60,60 @@ const Dropdown = forwardRef(
     const containerRef = useRef(null)
     const dropdownRef = useRef(null)
 
+    const dropdownWrapperStyles = useStyles({
+      [styles['dropdown']]: true,
+      [styles['dropdown__wrapper']]: true,
+      [styles[`dropdown__wrapper--${alignment}`]]: alignment
+    })
+
+    const dropdownWrapperIconStyles = useStyles({
+      [styles['dropdown__wrapper']]: true,
+      [styles['dropdown__wrapper__icon']]: true
+    })
+
+    const dropdownBodySizeStyles = useStyles(
+      {
+        [styles['dropdown']]: true,
+        [styles[`dropdown--${size}`]]: size,
+        [styles['dropdown__wrapper__icon']]: true,
+        [styles['dropdown--hidden']]: isEmpty(style)
+      },
+      {
+        className
+      }
+    )
+
+    const dropdownBodyStyles = useStyles({
+      [styles['dropdown__body']]: true
+    })
+
+    const dropdownLabelStyles = useStyles({
+      [styles['dropdown__label']]: true,
+      [styles['dropdown__label--placeholder']]: asPlaceholder
+    })
+
+    const dropdownStyles = useStyles(
+      {
+        [styles['dropdown__wrapper']]: !custom,
+        [styles[`dropdown__wrapper--${alignment}`]]: alignment,
+        [styles['dropdown__wrapper--disabled']]: isOpenDisabled,
+        [styles['dropdown__wrapper--input']]: hasInput,
+        [styles['dropdown__wrapper--as-input']]: hasFullInputStyle,
+        [styles['dropdown__wrapper--input--disabled']]:
+          hasInput && isOpenDisabled
+      },
+      {
+        className
```

Comment:
j.w, dodatkowo to jest jeden props className a jest rozbity na style dla dwóch rzeczy, przekazując sobie jakiś className klasy zostaną dodane w dwóch miejscach takie same.